### PR TITLE
Search single product

### PR DESCRIPTION
**Pull request by BelideNikhil**
### Features

+ Search Products Functionality
+ Single Product card

### CODE REVIEW

File: src/Components/Navbar/NavSearch.jsx [hunk 1]
```diff
@@ -1,8 +1,34 @@
...
+import { useProduct } from "../../Context/ProductContext";
+const { RESET_PRODUCT_FILTERS } = actionTypes;
+export default function NavSearch() {
+    const { productDispatchFun } = useProduct();
+    const currentPage = useLocation();
+    const navigate = useNavigate();
+
+    function searchProductsFunction(e) {
+        productDispatchFun({ type: RESET_PRODUCT_FILTERS });
+        productDispatchFun({ type: "SEARCH_PRODUCTS", payload: { searchValue: e.target.value } });
```
Comment (himadri2110) [inline src/Components/Navbar/NavSearch.jsx:13]: Please stay consistent on the usage of action types.

**Review by himadri2110 — APPROVED** [review]
Good to go. The single Product page looks awesome.

File: src/Components/ProductCard/ProductCard.jsx [hunk 2]
```diff
@@ -24,17 +23,27 @@ export default function ProductCard({ card }) {
...
     return (
-        <div className={`product-card card card-with-icon card-with-badge  ${outOfStock ? "card-overlay" : ""}`}>
+        <div
+            className={`product-card card card-with-icon card-with-badge  ${
+                outOfStock ? "card-overlay out-of-stock" : ""
+            } `}
+            onClick={() => navigate(`${card.id}`)}
+        >
             <div className="card-image">
                 <img src={source} alt={alt} />
```
Comment (Rohanmond) [inline src/Components/ProductCard/ProductCard.jsx:33]: give better alt name

**Review by Rohanmond — APPROVED** [review]
code looks good to merge

File: src/Components/ProductCard/ProductCard.jsx (hunk 2 above)
Comment (BelideNikhil) [reply src/Components/ProductCard/ProductCard.jsx:33]: the alt is dynamic

**Pull request merged**